File: src/easy/#557. Reverse Words in a String III.js
// https://leetcode.com/problems/reverse-words-in-a-string-iii/description/

function reverseWords(s) {
  const arr = s.split(" ");
  let res = "";

  for (let i = 0; i < arr.length; i++) {
    for (let j = arr[i].length - 1; j >= 0; j--) {
      res = res + arr[i][j];
    }

    res = res + " ";
  }

  return res.trim();
}

function reverseWords(s) {
  const arr = s.split(" ");

  return arr.map((s) => s.split("").reverse().join("")).join(" ");
}

File: src/easy/#557. Reverse Words in a String III.py
def reverseWords(s: str) -> str:
    arr = s.strip().split(' ')

    return " ".join(list(map(lambda word: ''.join(reversed(word)), arr)))


def reverseWords(s):
    arr = s.split(' ')
    res = ''

    for i in range(len(arr)):
        res = res + arr[i][::-1]+' '

    return res.strip()
